lib/diagnostics_channel.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const {
+  ObjectDefineProperty,
+  ObjectSetPrototypeOf,
+  ObjectValues,
+} = primordials;
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+  }
+} = require('internal/errors');
+
+// TODO(qard): should there be a C++ publisher interface?
+class ActivePublisher {
+  subscribe(subscription) {
+    if (typeof subscription === 'function') {
+      this._subscribers.push(subscription);
+      return;
+    }
+
+    if (typeof subscription === 'object' &&
+        typeof subscription.onMessage === 'function') {
+      let shouldListen = true;
+
+      if (typeof subscription.shouldListen === 'function') {
+        shouldListen = subscription.shouldListen(this.name);
+      }
+
+      if (shouldListen) {
+        this._subscribers.push(subscription.onMessage.bind(subscription));
+      }
+    }
+  }
+
+  shouldPublish() {
+    return true;
+  }
+
+  publish(data) {
+    for (var i = 0; i < this._subscribers.length; i++) {
+      this._subscribers[i](this.name, data);
+    }
+  }
+}
+
+class Publisher {
```

Comment:
Avoids potential Object.prototype setters
```suggestion
class Publisher {
_subscribers;
name;
```

Reply:
I tried this before, but I was getting acorn-related errors from `make test-doc`.

Reply:
this file shouldn't be firing for test-doc to my knowledge. Either way, right now Object.prototype setters are being fired if they exist and an own field should be defined rather than normal assignment.

Reply:
Seems to for me. If I run `make test` it gets to the `test-doc` part and blows up, pointing to those exact lines in an acorn syntax error. 🤷

Reply:
Appears to specifically be `tools/doc/apilinks.js` which breaks this. Seems to break on a bunch of types of new syntax.

Reply:
Additional observation: this seems to only apply to top-level modules. Anything in `lib/internal` can use the new syntax without running into this issue. We seem to just not yet have any instances of this syntax in top-level modules yet. 🤷